cli/cppcheckexecutor.cpp

```diff
@@ -310,11 +303,15 @@ int CppCheckExecutor::check_internal(CppCheck& cppcheck)
         AnalyzerInformation::writeFilesTxt(settings.buildDir, fileNames, settings.userDefines, settings.project.fileSettings);
     }
 
+    CppCheck cppcheck(*this, true, executeCommand);
+    cppcheck.settings() = settings; // this is a copy
```

Comment:
This highlights the issue I want to resolved by moving the ownership output `CppCheck`. We need to propagate back all the changes from the settings attached to `CppCheck`. In some cases we are not doing this correctly. This is why the settings should be immutable during the analysis so we don't have to do that.

Also we do that copy for *each* file with the multi-threaded implementation which is a hot spot if there are many libraries loaded.